viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
 
-        
+
+def get_unique_watched(user_data):
+    user_unique_watched = []
+    user_watched = []
+    friends_watched = []
+
+    for movies in user_data["watched"]:
+        user_watched.append(movies)
+
+    for movies in user_data["friends"]:
+        for details in movies["watched"]:
+            friends_watched.append(details)
+
+    user_unique_watched = [i for i in user_watched if i not in friends_watched]
+
+    if len(user_unique_watched) == []:
+        return 0
+    else:
+        return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+    user_watched = []
+    friends_watched = []
+
+    for movies in user_data["watched"]:
+        user_watched.append(movies)
+
+    for movies in user_data["friends"]:
+        for details in movies["watched"]:
+            friends_watched.append(details)
+
+    friends_unique_watched = [
+        i for i in friends_watched if i not in user_watched]
+
+    no_duplicates_movie_list = []
+    for i in friends_unique_watched:
+        if i not in no_duplicates_movie_list:
+            no_duplicates_movie_list.append(i)
+
+    if len(friends_unique_watched) == []:
+        return 0
+    else:
+        return no_duplicates_movie_list
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommend_list = []
+    combined_list = []
+    subscriptions = []
+    for movies in user_data["subscriptions"]:
+        subscriptions.append(movies)
+
+    friends_list = get_friends_unique_watched(user_data)
+    users_list = get_unique_watched(user_data)
+    combined_list = friends_list + users_list
+
+    for item in friends_list:
+        for details in item:
+            if item["host"] in subscriptions:
+                recommend_list.append(item)
+
+    newlist = []
+    for i in recommend_list:
+        if i not in newlist:
+            newlist.append(i)
+
+    if len(newlist) == []:
+        return 0
+    else:
+        return newlist
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    new_rec_genre = []
+    friends_list = get_friends_unique_watched(user_data)
+    most_watched_genre = get_most_watched_genre(user_data)
+    for movie in friends_list:
+        if movie["genre"] == most_watched_genre:
+            new_rec_genre.append(movie)
+    if len(new_rec_genre) == []:
+        return 0
+    else:
+        return new_rec_genre
+
+
+def get_rec_from_favorites(user_data):
+    recommend_list = []
+    user_favorite = []
+    user_unique_list = get_unique_watched(user_data)
+
+    for movie in user_data["favorites"]:
+        user_favorite. append(movie)
+
+    for movie in user_unique_list:
+        if movie in user_favorite:
+            recommend_list.append(movie)
+    return recommend_list

```

Comment:
We could combine some of these by using `get_available_recs` and then loop through them, then check against user's favorites!

```suggestion
    recommend_list = get_available_recs(user_data)
    fav_rec_list = []

   for movie in recommend_list:
        if movie in user_data["favorites"]:
            fav_rec_list.append(movie)
    
    return fav_rec_list
```